crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1051,6 +1051,14 @@ impl FStringValue {
         matches!(self.inner, FStringValueInner::Concatenated(_))
     }
 
+    /// Returns a slice of all the [`FStringPart`]s contained in this value.
+    pub fn as_slice(&self) -> &[FStringPart] {
```

Comment:
Do we still need the `parts` method` or can we replae it with `as_slice` (same for `parts_mut` that we could replace with `as_slice_mut`?

Reply:
We could which will require updating all the references to use `as_slice().iter()`. I think it should be a quick change. I'll do it before merging this PR, thanks for the suggestion.

Reply:
We can also rename the methods to `iter` and return `std::slice::Iter` instead of `impl Iterator`. It would then make sense to possibly also implement `IntoIterator` (which is common for types implementing `as_slice()` and `iter`)